libdeflate: Update to v1.17

oliv3r opened the pull request:
CMake depends on (libdeflate-)gunzip, libdeflate depends on Cmake, so we can't win.

Luckily libdeflate is _very_ easy to build, without any build system, so lets just manually compile it and be done with it.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

oliv3r committed:
libdeflate: Avoid circular dependencies

CMake depends on (libdeflate-)gunzip, libdeflate depends on Cmake, so we
can't win.

Luckily libdeflate is _very_ easy to build, without any build system, so
lets just manually compile it and be done with it.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>
Signed-off-by: Rosen Penev <rosenp@gmail.com>

neheb commented on tools/Makefile:
```diff
@@ -89,7 +89,7 @@ $(curdir)/genext2fs/compile := $(curdir)/libtool/compile
 $(curdir)/gmp/compile := $(curdir)/libtool/compile
 $(curdir)/isl/compile := $(curdir)/gmp/compile
-$(curdir)/libdeflate/compile := $(curdir)/cmake/compile
+$(curdir)/libdeflate/compile := $(curdir)/libdeflate/compile
```
uhhh what?

oliv3r replied:
I have no idea :D but would appreciate some help here. I was looking at the other stuff above. basically, we only depend on gccc ...

neheb replied:
delete this line. libdeflate is not dependent on anything after this change.

neheb commented on tools/libdeflate/Makefile:
```diff
@@ -16,7 +16,9 @@ PKG_SOURCE_VERSION:=v$(PKG_VERSION)
-include $(INCLUDE_DIR)/cmake.mk
+
+define Host/Compile
+	$(CC) lib/*{,/*}.c programs/{gzip,prog_util,tgetopt}.c -o libdeflate-gzip
```
HOST_CFLAGS should also be passed.

oliv3r commented:
@neheb btw, how can I force a rebuild of a tool? It keeps telling me 'everything is up to date'. `make tools/libdeflate` with clean, compile, anything or everything, `make oldconfig` nothing yields a rebuild ... (i tried searching for libdeflate stamps and artifacts, but found nothing to delete anymore. ...

robimarko commented:
You tried changing the `PKG_RELEASE`?

neheb commented:
I think it's because this package is using Host/Clean instead of Host/Uninstall

Ansuel commented:
Oh wow it can simply be compiled with good old gcc raw command? Nice!

oliv3r commented:
> You tried changing the `PKG_RELEASE`?

Part of the Merge Request already :) but no :( doesn't work.

oliv3r commented:
> I think it's because this package is using Host/Clean instead of Host/Uninstall
> 
> edit: yeah, need to call Host/Clean/Default in that section.

So this goes beyond my knowledge (and desire to know) about the OpenWRT build system :p however, no, doesn't seem to make a difference. I tried several combinations, Host/Uninstall, Host/Clean; adding the call (I did push that bit as I saw other tools having it). But a) I have no idea how all of this magic works, and b) I can't force a rebuild to check if it actually compiles. So feel free to merge it on good faith :) or test it somehow. I can't seem to :(

neheb commented:
I'll fix it in the other PR.

oliv3r commented:
@neheb thank you for fixing it in your MR, i'll close this and refer to yours!

oliv3r committed:
libdeflate: Update to v1.17

The new version of libdeflate makes it a little easier to build it
without any build system.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

oliv3r commented:
So I figured things here, which due to a version bump makes building even easier.

oliv3r commented:
@neheb should be all good now I hope :)

neheb commented on tools/libdeflate/Makefile:
```diff
@@ -8,24 +8,32 @@ include $(TOPDIR)/rules.mk
+	$(HOSTCC_NOCACHE) $(HOST_CFLAGS) $(HOST_LDFLAGS) \
+	$(HOST_BUILD_DIR)/lib/*{,/*}.c \
+	$(HOST_BUILD_DIR)/programs/{gzip,prog_util,tgetopt}.c \
+	-I$(HOST_BUILD_DIR) \
```
Remove

oliv3r replied:
Everything? or just 'programs'?

neheb replied:
The -I line is not needed.

oliv3r replied:
ah, ok fair nuff!

neheb commented:
Change it to libdeflate: update to 1.17

neheb commented on tools/Makefile:
```diff
@@ -136,8 +134,9 @@ $(foreach tool, $(tools-y), $(if $(wildcard $(curdir)/$(tool)/patches),$(eval $(
-tools-core := tar xz patch
+# make any tool depend on libdeflate, patch, tar, and xz to ensure that archives
+# can be unpacked and patched properly
+tools-core := libdeflate patch tar xz
```
Doesn't belong in this commit. You're not changing the unpack routine.

oliv3r replied:
Are you telling me order matters? You put libdeflate as part of the tools-core, so I took that over, and fixed sorting as part of this.

Ansuel replied:
should we change this list like the tools one (one tool per line) for consistency and better tracking?

neheb replied:
no. this PR doesn't force unpacking with libdeflate. why would it be moved to tools-core?

oliv3r replied:
That's something I have no idea for; but I took this from your set of changes @neheb

neheb commented:
Change it to libdeflate: update to 1.17 and keep second commit title lowercase.

oliv3r commented:
> Change it to libdeflate: update to 1.17 and keep second commit title lowercase.

"Officially" Linux and in many git guidelines, you are supposed to start your subject with a capital. I'll make it consistent.

neheb commented on tools/libdeflate/Makefile:
```diff
@@ -23,7 +23,7 @@ define Host/Install
-define Host/Clean
+define Host/Uninstall
 	rm -f $(STAGING_DIR_HOST)/bin/libdeflate-gzip
 	rm -f $(STAGING_DIR_HOST)/bin/libdeflate-gunzip
```
you're missing Uninstall/Default

neheb replied:
nvm. next commit adds it.

Ansuel commented:
@oliv3r also can you change `(actually fix the PR so that it works)` to [ ... ] ? 

With the 2 requeste change, I will merge this and then I will merge https://github.com/openwrt/openwrt/pull/11717 rebased

oliv3r committed:
tools: Improve diffability/maintainability

Like with commit ae614fb397c2 ("tools: Improve diffability/maintainability")
we also want tools-core to be easy to maintain. While a smaller target,
it's still usefull and makes things nice and consistent.

To avoid duplicating any tools in the comment, simplify the comment
instead.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

oliv3r commented:
> @oliv3r also can you change `(actually fix the PR so that it works)` to [ ... ] ?
> 
> With the 2 requeste change, I will merge this and then I will merge #11717 rebased

Done! Lets get this puppy in!

Ansuel commented:
@oliv3r missing separator error

Ansuel commented:
There are some rebase comment left

Ansuel commented on tools/Makefile:
```diff
@@ -98,8 +97,11 @@ $(curdir)/genext2fs/compile := $(curdir)/libtool/compile
 $(curdir)/gengetopt/compile := $(curdir)/libtool/compile
 $(curdir)/gmp/compile := $(curdir)/libtool/compile
 $(curdir)/isl/compile := $(curdir)/gmp/compile
+<<<<<<< HEAD
```
here

oliv3r replied:
yeah, no idea how that snook in ... 0bviously during rebase :p

Ansuel commented:
@oliv3r just in case this is rebased on top of master so i can merge without removing your verified tag?

oliv3r commented:
> @oliv3r just in case this is rebased on top of master so i can merge without removing your verified tag?

Not sure what you meant, but it should have been rebased on master (as I always do a fetch; rebase; cycle), but since it's merged, I guess its good :)

Ansuel commented:
(to give you some context... to flag pr as merged they needs to be merged with --ff-only so they have to be rebased on top of master... and by rebasing on top of master i would have dropped your verified tag as i would had to resign them for the force push... but they were rebased so np :D)
